### PR TITLE
Handling for bad search requests

### PR DESCRIPTION
**Pull request by christophermancini**
I added code to abort the processing of the result set when a 400 status code has been returned to account for issue #103 (CLIENTS-597). Added a new test for this case.

### CODE REVIEW

**Comment by christophermancini** [comment]
@karlra let me know if this is an acceptable solution for you.

File: src/Riak/Command/Search/Response.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct($statusCode, $headers = [], $body = '')
         parent::__construct($statusCode, $headers, $body);
 
         // make sure body is not only whitespace
-        if (trim($body)) {
+        if (trim($body) && $this->statusCode != 400) {
```
Comment (javajolt) [inline src/Riak/Command/Search/Response.php]: Are there other error status codes you want to exclude here?  Or simply say `$this->statusCode == 200`?

Comment (lukebakken) [inline src/Riak/Command/Search/Response.php]: ^^^^ what @javajolt said, easier to test for success here, I think. 200, 201, 204, maybe?

I have this reference bookmarked: http://www.restapitutorial.com/httpstatuscodes.html

**Comment by christophermancini** [comment]
@lukebakken @javajolt Updated code as we discussed offline.

**Comment by lukebakken** [comment]
:ship: :it:

**Comment by christophermancini** [comment]
Tests are still passing post last commit. Merging!

**Pull request merged**